Add LCC and VSC models with PSS/E Parser

rodrigomha opened the pull request:
New PR that superseedes #1205. Too many conflicts with psy5 branch

jd-lara commented on src/PowerSystems.jl:
```diff
@@ -34,7 +34,8 @@ export Line
 export MonitoredLine
 export DCBranch
 export TwoTerminalHVDCLine
```
```suggestion
export TwoTerminalGenericHVDCLine
```

jd-lara reviewed (approved):
LGTM I suggested a small name change

GabrielKS commented on src/parsers/power_models_data.jl:
```diff
@@ -900,41 +902,146 @@ function read_3w_transformer!(
-function make_dcline(name::String, d::Dict, bus_f::ACBus, bus_t::ACBus)
-    return TwoTerminalHVDCLine(;
+function make_dcline(name::String, d::Dict, bus_f::ACBus, bus_t::ACBus, source_type::String)
+    if source_type == "pti"
```
Could use `Val` to dispatch on `source_type`

rodrigomha replied:
It's possible but an overkill IMO, so we should be fine

GabrielKS commented on src/parsers/power_models_data.jl:
```diff
@@ -900,41 +902,146 @@ function read_3w_transformer!(
+            active_power_flow = get(d, "pf", 0.0),
+            r = d["r"],
+            transfer_setpoint = d["transfer_setpoint"],
+            scheduled_dc_voltage = d["scheduled_dc_voltage"],
```
Could do some `kwargs...` splatting to streamline this arguments list

rodrigomha replied:
... maybe next time?

GabrielKS commented on src/parsers/powerflowdata_data.jl:
```diff
@@ -789,6 +789,7 @@ function read_dcline!(
     bus_number_to_bus::Dict{Int, ACBus};
     kwargs...,
 )
+    # TODO: We should be able to get this PowerFlowData LCC lines working in future work.
```
Add an issue for this?

rodrigomha replied:
Added #1272

GabrielKS commented on test/test_data/lcc_test.raw:
I think this exists as `two-terminal-hvdc_test.raw`  in PowerSystemsTestData, can we use that?

rodrigomha replied:
Done

GabrielKS reviewed (commented):
A few minor comments.

rodrigomha commented:
> LGTM I suggested a small name change

Updated with the new name, and also added deprecation for older name so we don't need to update TestData or CaseBuilder for now.

rodrigomha commented:
Tests are passing locally, but they are failing in some obscure `ext` thing in 2w transformers.

I think we should merge to 3WTransformer, update everything there, and merge to psy5 asap.
